test/handlers/FlexVotingClientHandler.sol

```diff
@@ -0,0 +1,365 @@
+    _;
+  }
+
+  modifier useActor(uint256 actorIndexSeed) {
+    currentActor = _randAddress(actors, actorIndexSeed);
+    _;
+  }
+
+  modifier useVoter(uint256 _voterSeed) {
+    currentActor = _randAddress(voters, _voterSeed);
+    _;
+  }
+
+  function hasPendingVotes(address _user, uint256 _proposalId) external view returns (bool) {
+    return pendingVotes[_proposalId].contains(_user);
+  }
+
+  function _randAddress(EnumerableSet.AddressSet storage _addressSet, uint256 _seed)
+    internal
+    view
+    returns (address)
+  {
+    uint256 len = _addressSet.length();
+    return len > 0 ? _addressSet.at(_seed % len) : address(0);
+  }
+
+  function getProposals() external view returns (uint256[] memory) {
+    return proposals.values();
+  }
+
+  function getVoters() external view returns (address[] memory) {
+    return voters.values();
+  }
+
+  function getActors() external view returns (address[] memory) {
+    return actors.values();
+  }
+
+  function lastProposal() external view returns (uint256) {
+    if (proposals.length() == 0) return 0;
+    return proposals.at(proposals.length() - 1);
+  }
+
+  function lastActor() external view returns (address) {
+    if (actors.length() == 0) return address(0);
+    return actors.at(actors.length() - 1);
+  }
+
+  function lastVoter() external view returns (address) {
+    if (voters.length() == 0) return address(0);
+    return voters.at(voters.length() - 1);
+  }
+
+  function _randProposal(uint256 _seed) internal view returns (uint256) {
+    uint256 len = proposals.length();
+    return len > 0 ? proposals.at(_seed % len) : 0;
+  }
+
+  // forgefmt: disable-start
+  function _validActorAddress(address _user) internal view returns (bool) {
+    return _user != address(0) &&
+      _user != address(flexClient) &&
+      _user != address(governor) &&
+      _user != address(receiver) &&
+      _user != address(token);
+  }
+  // forgefmt: disable-end
+
+  function remainingTokens() public view returns (uint128) {
+    return MAX_TOKENS - ghost_mintedTokens;
+  }
+
+  function proposal(uint256 _index) public view returns (uint256) {
+    return proposals.at(_index);
+  }
+
+  function proposalLength() public view returns (uint256) {
+    return proposals.length();
+  }
+
+  function actorsLength() public view returns (uint256) {
+    return actors.length();
+  }
+
+  // TODO This always creates a new actor. Should it?
+  function deposit(uint208 _amount)
+    external
+    maybeCreateActor(_amount)
+    maybeCreateVoter
+    countCall("deposit")
+  {
+    vm.assume(remainingTokens() > 0);
+    _amount = uint208(_bound(_amount, 0, remainingTokens()));
+
+    // Some actors won't have the tokens they need. This is deliberate.
+    if (_amount <= remainingTokens()) {
+      token.exposed_mint(currentActor, _amount);
+      ghost_mintedTokens += uint128(_amount);
+    }
+
+    vm.startPrank(currentActor);
+    // TODO We're pre-approving every deposit, should we?
```

Comment:
I tried to call out the assumptions as I was making them. Feel free to push back if you think any of them were incorrect or too strong.